### PR TITLE
fest(ddl-shim): Improved IField Schema

### PR DESCRIPTION
**Pull request by GordonSmith**
Switch to new schema generator
Refactor code
Removed "validate" as not needed.
Fixed "validate2".

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: demos/gallery/.vscode/launch.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
             "name": "galleryItem",
             "type": "chrome",
             "request": "launch",
-            "url": "file:///${workspaceRoot}/galleryItem.html?./samples/codemirror/DOT Editor.js",
+            "url": "file:///${workspaceRoot}/galleryItem.html?./samples/codemirror/DDL Editor.js",
```
Comment (jbrundage) [inline demos/gallery/.vscode/launch.json]: I can't find anything online about a "DOL" programming language?

Comment (jbrundage) [reply demos/gallery/.vscode/launch.json]: Disregard... I see now that the 2nd letter is a "D"

**Comment by GordonSmith** [comment]
@drealeed how did the Schema -> Java generation go?  Is this ok to pull?

**Comment by drealeed** [comment]
The schema-to-java generation worked with the new IField definition types, but @lrenn had trouble converting some existing .22 ddls with it. Let me do a bit more testing.

**Comment by drealeed** [comment]
Our testing worked, you can go ahead and merge.

**Pull request merged**